submissions/Dovahkiin1991/classic-frogger-game/app.js

```diff
@@ -0,0 +1,132 @@
+//const
+const CELL_WIDTH = 101,
+    CELL_HEIGHT = 83,
+    PLAYER_SPRITE_EMPTY_SPACE = 19,
+    PLAYER_Y_CENTER_SHIFT = 32,
+    ENEMY_Y_CENTER_SHIFT = 23,
+    ENEMIES_COUNT = 3,
+    allEnemies = [];
+
+const Enemy = function(enemyRowPosition, player) {
+    this.sprite = 'images/enemy-bug.png';
+    this.x = -CELL_WIDTH;
+    this.y = CELL_HEIGHT - ENEMY_Y_CENTER_SHIFT + enemyRowPosition * CELL_HEIGHT;
+    this.speed = speedGenerator();
+    this.player = player;
+};
+
+Enemy.prototype.update = function(dt) {
+    this.x += dt * this.speed;
+
+    if (this.x > CELL_WIDTH * 5) {
+        this.x = -CELL_WIDTH;
+        this.speed = speedGenerator();
+    }
+
+    this.collisionsCheck(player);
```

Comment:
`player` in this method is a global variable.
See item 4.iii of the [requirements](https://kottans.org/documentation/docs/doc/code-review/#oo-js-frogger).